### PR TITLE
Added support to create credentials, install plugins

### PR DESCRIPTION
**Pull request by skdeep12**
Added support to create secret text credentials, username password credential. Also, to install plugins.

### CODE REVIEW

File: credentials.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	if _, err := jenkins.checkCrumb(req); err != nil {
+		return err
+	}
+
+	resp, err := jenkins.sendRequest(req)
+	if err != nil {
+		return
+	}
+	if !(200 <= resp.StatusCode && resp.StatusCode <= 299) {
+		return errors.New(fmt.Sprintf("error: HTTP POST returned status code %d (expected 2xx)", resp.StatusCode))
```
Comment (yosida95) [inline credentials.go]: You can use `fmt.Errorf` here.

File: credentials.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	jenkins.createCredentials(header, jsonPayload)
+}
+
+func (jenkins *Jenkins) createCredentials(header map[string]string, jsonPayload string) {
+	form := url.Values{"json": []string{jsonPayload}}
+
+	var body interface{}
+	err := jenkins.postForm("/credentials/store/system/domain/_/createCredentials", header, url.Values{}, form, body)
+	if err != nil {
+		fmt.Println("Error: " + err.Error())
```
Comment (yosida95) [inline credentials.go]: We should do proper error handling and propagate an error to the method caller.

File: credentials.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	} else {
+		fmt.Println(body)
```
Comment (yosida95) [inline credentials.go]: We shouldn't print the response body to the stdout (I'm not sure about this endpoint, but might contains the secrets?).

File: plugins.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+	payload := `<jenkins><install plugin="` + name + `@` + version + `" /></jenkins>`
+
+	xmlBody := strings.NewReader(payload)
+
+	var body interface{}
+	err := jenkins.postXml("/pluginManager/installNecessaryPlugins", url.Values{}, xmlBody, &body)
+	if err != nil {
+		return err
+	}
+	fmt.Println(body)
```
Comment (yosida95) [inline plugins.go]: Again, we shouldn't print the response to the stdout.

File: credentials.go [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	}
+	if _, err := jenkins.checkCrumb(req); err != nil {
+		return err
+	}
+
+	resp, err := jenkins.sendRequest(req)
+	if err != nil {
+		return
+	}
+	if !(200 <= resp.StatusCode && resp.StatusCode <= 299) {
```
Comment (yosida95) [inline credentials.go]: Can you narrow this condition? In the success case, do we have to expect the 200-299 status code? What about 204 Not Content?

File: credentials.go (hunk 1 above)
Comment (skdeep12) [reply credentials.go]: used.
https://github.com/yosida95/golang-jenkins/blob/1b8eb2ef03ecce03532d9ae54b2497325b01e666/credentials.go#L68-L70

File: credentials.go (hunk 2 above)
Comment (skdeep12) [reply credentials.go]: I have improved the error handling.
https://github.com/yosida95/golang-jenkins/blob/1b8eb2ef03ecce03532d9ae54b2497325b01e666/credentials.go#L95-L97

File: credentials.go (hunk 3 above)
Comment (skdeep12) [reply credentials.go]: removed the print statements.

File: credentials.go (hunk 5 above)
Comment (skdeep12) [reply credentials.go]: In the case of a new secret creation or an existing secret update request, Jenkins returns  200 code only.
https://github.com/yosida95/golang-jenkins/blob/1b8eb2ef03ecce03532d9ae54b2497325b01e666/credentials.go#L68-L69

File: plugins.go (hunk 4 above)
Comment (skdeep12) [reply plugins.go]: removed.

**Pull request closed without merging**